feat(chart): add container security context (#31043)
mistercrunch/Airflow: docs/helm-chart/production-guide.rst

  * :ref:`uid <parameters:Airflow>` (configures the global uid or RunAsUser)
  * :ref:`gid <parameters:Airflow>` (configures the global gid or fsGroup)

>   * :ref:`securityContext <parameters:Kubernetes>` (deprecated, use ``securityContexts`` instead.)

Reviewer: ```suggestion
```

We don't have to mention the old way :)

  * :ref:`securityContexts <parameters:Kubernetes>` (same as ``uid`` but allows for setting all `Pod securityContext options <https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#podsecuritycontext-v1-core>`_ and `Container securityContext options <https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#securitycontext-v1-core>`_)

The same way one can configure the global :ref:`securityContexts <parameters:Kubernetes>`, it is also possible to configure different values for specific workloads by setting their local ``securityContexts`` as follows:

.. code-block:: yaml

  workers:

>     securityContext:
>     pod:
>       runAsUser: 5000
>       fsGroup: 0
>     containers:
>       allowPrivilegeEscalation: false

Reviewer: ```suggestion
    securityContexts:
      pod:
        runAsUser: 5000
        fsGroup: 0
      containers:
        allowPrivilegeEscalation: false
```

Right?

> In the example above, the workers Pod ``securityContexts`` will be set to ``runAsUser: 5000`` and ``runAsGroup: 0``.  The containers pod will be set to ``allowPrivilegeEscalation: false``.

Reviewer: ```suggestion
In the example above, the workers Pod ``securityContexts`` will be set to ``runAsUser: 5000`` and ``fsGroup: 0``.  The containers pod will be set to ``allowPrivilegeEscalation: false``.

Existing error! Oops!
```

Author: Suggestion not committed as is because `Existing error! Oops!` was included in by mistake. It's fixed nevertheless

Reviewer: Oh my bad 🙃

  * :ref:`uid <parameters:Airflow>` (configures the global uid or RunAsUser)
  * :ref:`gid <parameters:Airflow>` (configures the global gid or fsGroup)
  * :ref:`securityContext <parameters:Kubernetes>` (same as ``uid`` but allows for setting all `Pod securityContext options <https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#podsecuritycontext-v1-core>`_)
  * :ref:`securityContexts <parameters:Kubernetes>` (same as ``securityContext`` with additional security context on the container level `Container securityContext options <https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#securitycontext-v1-core>`_)

> The same way one can configure the global :ref:`securityContext <parameters:Kubernetes>`, it is also possible to configure different values for specific workloads by setting their local ``securityContext`` as follows:

Reviewer: Since we are deprecating `securityContext`, we need to update this example to show `securityContexts` instead.
